### PR TITLE
Fixed Issue #25227:Issue with Product Review when disabled from admin

### PR DESCRIPTION
**Pull request by ghost**
### Fixed Issues (if relevant)

1. magento/magento2#25227: Issue with Product Review when disabled from admin

### Manual testing scenarios (*)

1. Open Product page from frontend and go to add review section 
2. Now login into Admin.
3. Disable Product Review from admin. Navigate to **Store > Configuration** then **Catalog > Catalog** now disable **Product Reviews**
4. Now go to opened product page on frontend and submit review
5. Now issue is fixed. Now there will be no route page.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ghost** [comment]
@akaplya - Please review this Pull request.

**Comment by ghost** [comment]
@akaplya - Did you get the chance to review this PR?
Please review this and let me know if anything is required from my end.
Thanks

**Comment by ghost** [comment]
@akaplya - Please let me know the status of this PR.

File: app/code/Magento/Review/Controller/Product.php [hunk 1]
```diff
@@ -221,13 +245,13 @@ protected function loadProduct($productId)
...
-                throw new NoSuchEntityException();
+                return false;
             }
 
             if (!$product->isVisibleInCatalog() || !$product->isVisibleInSiteVisibility()) {
-                throw new NoSuchEntityException();
+                return false;
             }
-        } catch (NoSuchEntityException $noEntityException) {
+        } catch (\Exception $e) {
```
Comment (novikor) [inline app/code/Magento/Review/Controller/Product.php:254]: Why catching general` \Exception` instead of `NoSuchEntityException`?

**Review by novikor — CHANGES_REQUESTED** [review]
Hi, @rahulwebkul ! Thank you for your contribution.
Please check my comment

**Comment by novikor** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. Please feel free to reopen if you are going to continue working on it.
Thank you.

**Pull request closed without merging**